### DISCUSSION TITLE
Running `brew create --cask <new-cask-token>` results in Error

### DISCUSSION THREAD

**Discussion by cesarcoatl** [Homebrew/discussions#301] [Casks] [answered]
Hello.

I am attempting to add a new Cask under `homebrew-cask-versions` following the [Adding a Cask guide](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md), but when I run the [`brew create --cask`](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#the-brew-create---cask-command) command I get the following:

```bash
$ brew create --cask my-new-cask
...
Error: Invalid usage: The `--set-name` flag is required for creating casks.
```

After a couple of attempts, the only command that worked was:

```bash
$ brew create --cask my-new-cask --set-name my-new-cask
```

This last command did open my `$EDITOR` and allowed me to save my new cask into `"$(brew --repository)"/Library/Taps/homebrew/homebrew-cask/Casks`.

Just sharing this in case the documentation needs to be modified or the command behavior.

Thanks.

**Comment by hyuraku** [comment, marked as answer]
Thanks a lot @thecesrom 
I opened PR https://github.com/Homebrew/homebrew-cask/pull/95740 to fix this

**Reply by cesarcoatl** [reply]
As I mentioned, I was not sure if the documentation needed to be updated or the command behavior.

But now that I've read @SMillerDev's comments and having tried that, I have a few questions.

For the sake of clarity and consistency, specially with the [Basic Instructions](https://docs.brew.sh/Formula-Cookbook#basic-instructions) from the Formula Cookbook, wouldn't it be better to have the following?

> ### The `brew create URL --cask` Command
> 
> Once you know the token, create your Cask with the handy-dandy `brew create --cask` command:
> 
> ```bash
> $ brew create URL --cask --set-name my-new-cask
> ```
> 
> This will open `$EDITOR` with a template for your new Cask, to be stored in the file `my-new-cask.rb`. Running the `create` command above will get you a template that looks like this:
> 
> ```ruby
> cask "my-new-cask" do
>   version ""
>   sha256 ""
> 
>   url "URL"
>   name ""
>   desc ""
>   homepage ""
> 
>   app ""
> end
> ```

Additionally, I see another possible change, and that is to the help output by the `brew create --help` command. Since the URL is required, and the options, well, optional.

Change this:
```bash
$ brew create --help
Usage: brew create [options] URL
...
```
For this:
```bash
$ brew create --help
Usage: brew create URL [options]
...
```

**Reply by gromgit** [reply]
By convention, Unix commands take `-`-prefixed options immediately after the command name and before command arguments, exactly as it stands right now, so I don't see your suggestion being adopted any time soon.

**Reply by cesarcoatl** [reply]
Understood. Thanks.
